src/clean.js

```diff
@@ -0,0 +1,93 @@
+const { contentfulManagementClient, contentfulPreviewClient, defaultLocale } = require('./config');
+
+const deleteEntry = async(id) => {
+  const contentfulManagementConnection = await contentfulManagementClient.connect();
+  let entry = await contentfulManagementConnection.getEntry(id).catch((e) => {
+    console.log(`couldn't load entry: ${id}\n` + e.toString());
+    return;
+  });
+
+  if (entry.sys.publishedVersion) {
+    console.log(`unpublish ${entry.sys.contentType.sys.id}`);
+    entry = await entry.unpublish().catch((e) => {
+      console.log('error unpublishing ' + JSON.stringify(e));
+    });
+  }
+
+  console.log(`delete ${entry.sys.contentType.sys.id}`);
+  entry.delete().catch(async(e) => {
+    console.log('error deleting'  + JSON.stringify(e));
+    throw `Failed on ${id}`;
+  });
+};
+
+// https://stackoverflow.com/a/46842181/738371
+async function asyncFilter(arr, callback) {
```

Comment:
Remove this. See comment below for rewrite of other function without needing it.